### PR TITLE
Minimax

### PR DESCRIPTION
**Pull request by nadiacollado**
adds easy and advanced strategy classes

### CODE REVIEW

File: lib/game_config.rb [hunk 1]
```diff
@@ -25,7 +25,12 @@ def humans_only?
     
     def computer_selection
         display.print_computer_type_prompt
-        Computer.new(1, marker.p1_marker, display.get_computer_type)
+        if display.get_computer_type == EASY_COMPUTER
```
Comment (proxycase) [inline lib/game_config.rb:28]: A ternary might be a cleaner way to write this:
```
strategy = display.get_computer_type == EASY_COMPUTER ? EasyStrategy.new : AdvancedStrategy.new
```

File: lib/advanced_strategy.rb [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    def player_mark(maximazing)
+        maximazing ? AI : HUMAN
+    end
+
+    def make_integer(square)
+        square.to_i
+    end
+end
+
+# USES NEW INSTANCE OF BOARD. DOES NOT WORK.
```
Comment (proxycase) [inline lib/advanced_strategy.rb:67]: This is a pretty big chunk of commented out code. I'd document it a little better if you want to keep it in here. Include things like: What doesn't work about it, and why you're saving it for later.

**Review by proxycase — APPROVED** [review]
LGTM, just some suggestions!

File: lib/game_config.rb (hunk 1 above)
Comment (judan) [reply lib/game_config.rb:28]: +1

File: lib/advanced_strategy.rb (hunk 2 above)
Comment (judan) [reply lib/advanced_strategy.rb:67]: +1

**Review by judan — APPROVED** [review]
Nonblocking comments, but otherwise LGTM, especially tests: 

- Agree with's @proxycase's comments about adding refactoring notes. One thing to implement when you come back to this game would be to prompt the player to replay the game and run the game loop again / create a new game after one has ended.
- Wasn't sure how to unmark a square, or why that needs to be called in advanced mode. Is this a feature to implement later? (Add some notes)
- Overall good job implementing minimax, and just document things for when you pick this back up again.

Feel free to merge!

**Pull request merged**